### PR TITLE
Refactor, reduce, and refine tests.

### PR DESCRIPTION
**Pull request by LyonesGamer**
Fixes the mess that our tests are largely in right now, as a result of the rather natural evolution of the files and the lack of any consistent standard or guideline.

## Changes in this PR.

- [X] Cleans up times tests
- [x] Cleans up activities tests
- [x] Cleans up projects tests
- [x] Cleans up users tests

## Testing this PR.

1. Run tests (``npm run test``).
2. Read tests.

### Expected Output.

1. They still work just as well, and even a little faster.
2. Much of the boilerplate is removed, each test has a standardized form, and each file has one source of truth of test data (allowing easy changes later).

@osuosl/devs

### CODE REVIEW

File: tests/activities.js [hunk 1]
```diff
@@ -134,1520 +139,1028 @@ module.exports = function(expect, request, baseUrl) {
...
+          'deleted_at': null,
+          'updated_at': null,
+          'created_at': '2014-01-01',
+        },
+      ],
+    };
 
-          expect(jsonBody).to.deep.equal(expectedError);
-          expect(res.statusCode).to.equal(401);
+    beforeEach(function(done) {
```
Comment (ElijahCaine) [inline tests/activities.js]: This is confusing. Please comment it.

File: tests/activities.js [hunk 2]
```diff
@@ -134,1520 +139,1028 @@ module.exports = function(expect, request, baseUrl) {
...
-    // Performs get request to check whether the db's been changed
-    function checkGetReq(done, token) {
-      request.get(baseUrl + 'activities/docs?token=' + token,
-      function(getErr, getRes, getBody) {
-        expect(getErr).to.be.a('null');
-        expect(getRes.statusCode).to.equal(200);
-
-        const jsonBody = JSON.parse(getBody);
+    beforeEach(function(done) {
+      function getPostObject(uri, obj) {
```
Comment (ElijahCaine) [inline tests/activities.js]: This is declared in four beforeEach blocks, you might want to factor it out to somewhere else.

**Comment by ElijahCaine** [comment]
```
$ diff -y newtests.output oldtests.output
  378 passing (21s)                                           |   383 passing (21s)


  real    0m23.669s                                           |   real    0m24.243s
  user    0m24.640s                                           |   user    0m25.260s
  sys     0m1.320s                                                sys     0m1.320s
```
Very nice. I'm not sure where those extra 5 tests went, but this is a pretty good movement toward our tests being more maintainable.  +1 from me.

**Comment by LyonesGamer** [comment]
Those tests were ones that were extremely difficult to test given the forms of these tests, and which weren't very necessary anyway. (For example, testing that an endpoint returns a 401 if you request a token with a bad username/password; that's already covered in our login tests.)

**Comment by Kennric** [comment]
+1

**Pull request merged**